src/main/java/org/junitpioneer/jupiter/DisableIfNameExtension.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.junit.jupiter.api.extension.ConditionEvaluationResult.disabled;
+import static org.junit.jupiter.api.extension.ConditionEvaluationResult.enabled;
+import static org.junit.platform.commons.support.AnnotationSupport.findAnnotation;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.extension.ConditionEvaluationResult;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class DisableIfNameExtension implements ExecutionCondition {
+
+	@Override
+	public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext context) {
+		Optional<DisableIfName> disable = findAnnotation(context.getElement(), DisableIfName.class);
+
+		if (!disable.isPresent()) {
+			return enabled("No instructions to disable");
+		} else if (!context.getTestMethod().isPresent()) {
+			return enabled("Only disable at method level");
```

Comment:
We could disable on class level, too, right? Would that make sense?

Reply:
The problem with disabling at class level is that this will get disabled before the parameterized tests could get registered.
Hence, all tests will end up getting disabled since the test name will never be calculated for a parameterized test

Reply:
Yeah, that makes sense. And I just checked that the annotation hs `@Target({ ElementType.METHOD })`, so this should not happen anyways.